src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import { nanoid } from "nanoid";
+import PropTypes from "prop-types";
+import React, { ReactElement, useRef } from "react";
+
+import "./modal.scss";
+
+type Props = {
+  buttons?: ReactElement[];
+  children?: ReactElement;
+  close?: () => void;
+  title?: string;
```

Comment:
It could be nice to have this as ReactNode in case someone wants to display something else in this position other than just a string (e.g. an icon or a link). I tend to try and be as flexible as possible with these types.